distributed_shampoo/distributed_shampoo.py

```diff
@@ -229,16 +240,20 @@ class DistributedShampoo(torch.optim.Optimizer):
         dampening (float): Dampening parameter for momentum. (Default: 0.)
         weight_decay (float): Weight decay (L2 penalty). (Default: 0.)
         max_preconditioner_dim (int): Maximum preconditioner dimensio. (Default: 1024)
-        precondition_frequency (int): Frequency for computing root inverse preconditioner. (Default: 1)
+        precondition_frequency (int): Frequency for computing root inverse preconditioner. If use_eigenvalue_correction=True,
+            this is the frequency of the preconditioner eigenbasis computation. (Default: 1)
         start_preconditioning_step (int): Iteration to start computing inverse preconditioner. If -1, uses
             the same value as precondition_frequency. (Default: -1)
         inv_root_override (int, Sequence[int]): Inverse root to use in Shampoo. If a list [l1, l2, ..., lp], then we will
             use -1 / l1 for 1-D tensor (vectors), -1 / l2 for 2-D tensors (matrices), and so on. If the order of the
             tensor exceeds the order of the tensor, reverts to the default value. If 0 is used, uses the default inverse
-            root -1 / (2 * o), where o is the order of the tensor. (Default: 0)
+            root -1 / (2 * o), where o is the order of the tensor. If use_eigenvalue_correction=True, the default is -1 / 2.
+            (Default: 0)
         exponent_multiplier (float): Number to be multiplied to the numerator of the inverse root, i.e., eta where the
             exponent is -eta / (2 * p). (Default: 1.0)
-        use_nesterov (bool): Flag for using Nesterov momentum. (default: False)
+        use_eigenvalue_correction (bool): Flag for correcting the eigenvalues/running Adam in the eigenbasis of Shampoo's
```

Comment:
I think this functionality should be fully separate from the grafting configuration because it is in principle completely independent and algorithmically unrelated to grafting. However, we could log a warning in cases where we haven't really explored the eigenvalue correction yet, e.g. `use_eigenvalue_correction == True` and `betas[1] == 1.0` (AdaGrad-style accumulation).

Reply:
I still think that the eigenvalue correction should be kept independent from the grafting configuration. I can think more about an alternative interface to the current additional boolean flag that can be passed to the constructor.